force-app/main/default/classes/NextStepQueueable.cls

```diff
@@ -17,52 +17,70 @@
  * 
  * Note: The class below contains placeholders for implementing a queueable job to update accounts and opportunities.
  */
-public with sharing class NextStepQueueable {
+
+ public with sharing class NextStepQueueable implements Queueable {
 
     // Member variable to store the accounts
-    private Map<Id,Account> accounts;
+    private Map<Id, Account> accounts;
 
     // Constructor to accept the accounts
-    public NextStepQueueable(Map<Id,Account> accounts) {
-        // Store the accounts in a member variable
+    public NextStepQueueable(Map<Id, Account> accounts) {
+        this.accounts = accounts;
     }
 
-    // Method to queue the job
+    // Corrected method signature for enqueueing the job
     public static void enqueueJob() {
-        // Get 10 accounts with Next Step populated
+        List<Account> accountsToProcess = [SELECT Id, Next_Step__c FROM Account LIMIT 100];
+        
+        List<Account> filteredAccounts = new List<Account>();
+        for (Account acc : accountsToProcess) {
+            if (acc.Next_Step__c != null) {
+                filteredAccounts.add(acc);
+            }
+        }
 
-        // If there are any accounts, queue the job using System.enqueueJob
+        if (!filteredAccounts.isEmpty()) {
+            Map<Id, Account> accountMap = new Map<Id, Account>();
+            for (Account acc : filteredAccounts) {
+                accountMap.put(acc.Id, acc);
+            }
+            System.enqueueJob(new NextStepQueueable(accountMap));
+        }
     }
```

Comment:
This can be optimized. Quey accounts that the next steps are null and return it in a map.